taichi/transforms/demote_operations.cpp

```diff
@@ -14,6 +14,35 @@ class DemoteOperations : public BasicStmtVisitor {
   DemoteOperations() : BasicStmtVisitor() {
   }
 
+  void visit(BitExtractStmt *stmt) override {
+    // @ti.func
+    // def bit_extract(input, begin, end):
+    //   return (input >> begin) & ((1 << (end - begin)) - 1)
+
+    // https://github.com/taichi-dev/taichi/pull/1795#issuecomment-682367083
+    if (stmt->get_kernel()->program.config.async_mode)
+      return;
+
+    VecStatement statements;
+    auto one = statements.push_back<ConstStmt>(LaneAttribute<TypedConstant>(1));
+    auto begin = statements.push_back<ConstStmt>(
+        LaneAttribute<TypedConstant>(stmt->bit_begin));
+    auto esbeg = statements.push_back<ConstStmt>(
+        LaneAttribute<TypedConstant>(stmt->bit_end - stmt->bit_begin));
+    auto input_sar_begin = statements.push_back<BinaryOpStmt>(
+        BinaryOpType::bit_sar, stmt->input, begin);
+    auto one_shl_esbeg =
+        statements.push_back<BinaryOpStmt>(BinaryOpType::bit_shl, one, esbeg);
```

Comment:
nit: I would suggest calculating `((1LL << (stmt->bit_end - stmt->bit_begin)) - 1)` directly in the cpp file, just like which in `constant_fold`.